Documentation/ExtensionArchitecture/HowTo/Localization/Php.rst

```diff
@@ -18,6 +18,25 @@ Which method of localization to use depends on the current context:
 Localization in plain PHP
 =========================
 
+The :php:`$GLOBALS['LANG']` can be used to access the language service as
+instantiated :php:`\TYPO3\CMS\Core\Localization\LanguageService` class.
+The global :php:`LANG` array is available, if a backend user has been initialized,
+in particular in the following contexts:
+
+*   frontend: only if there is a logged-in backend user
```

Comment:
I don't know if the discussion about the various contexts is still relevant, if $GLOBALS['LANG'] is no longer used, but LanguageServiceFactory is used. We are still using the same LanguageService class and this seems to depend on the BE user (as before), but I am not entirely sure.

For the frontend we have TYPO3\CMS\Extbase\Utility\LocalizationUtility::translate and whatnot, but this is not mentioned here : https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ExtensionArchitecture/HowTo/Localization/Php.html#localization-in-frontend-context

I am moving the change to the globals to a separate [PR](https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/pull/2783) - I think that makes it easier to handle since this here seems to require some more research / discussion.

Reply:
LocalizationUtility is Extbase-dependent (see namespace), so should be used only in an Extbase context (although it works also in non-Extbase).

LanguageServiceFactory has three factory methods:
- create() -> give a locale
- createFromUserPreferences() -> hand over a backend user
- createFromSiteLanguage() -> pass a SiteLanguage entity (e.g. from the frontend request attribute).
So, the returned LanguageService is independently from the context.
https://github.com/TYPO3/typo3/blob/main/typo3/sysext/core/Classes/Localization/LanguageServiceFactory.php

Personally, I still use $GLOBALS['LANG'] in backend for my projects/extensions, as long as it is not officially deprecated. Either I have to use $GLOBALS['LANG'], or $languageServiceFactory->createFromUserPreferences() with the $GLOBALS['BE_USER'], so nothing is won at the end - just relying then on another global variable. At least as long as there is no backend user request attribute.

Reply:
@brotkrueml which changes exactly are you requiring? Or are they all resolved?

Reply:
@brotkrueml how can we proceed here?

Reply:
No answer since June 2023